pyOCD/target/memory_map.py

```diff
@@ -89,6 +89,8 @@ def containsAddress(self, address):
 
     def containsRange(self, start, end=None, length=None):
         if end is None:
+            if length is None:
+                raise ValueError("both end and length parameters are None")
```

Comment:
Good idea to check for both missing.  It might also be good to check if both are present and if so trigger an error.  An assert should make this easy.  If you want to leave it as a ValueError that works too.  The patch looks good as is.  Just wanted to mention it.
```
assert (end is not None) ^ (length is not None)
```